common_features.mk

```
@@ -342,6 +342,14 @@ ifeq ($(strip $(VELOCIKEY_ENABLE)), yes)
     SRC += $(QUANTUM_DIR)/velocikey.c
 endif
 
+ifeq ($(strip $(VIA_ENABLE)), yes)
+    DYNAMIC_KEYMAP_ENABLE := yes
+    RAW_ENABLE := yes
+    BOOTMAGIC_ENABLE := lite
+    SRC += $(QUANTUM_DIR)/via.c
+    OPT_DEFS += -DVIA_ENABLE
```

Just thinking about this, since there is a hard limit for the number of layers, it may be worth adding the `LAYER_STATE_8BIT` define here.  This switches to a uint8_t for the layer states, instead of a uint32_t, so it can save some space and time for the firmware. 

```suggestion
    OPT_DEFS += -DVIA_ENABLE
    OPT_DFES += -DLAYER_STATE_8BIT
```

---

There is no hard limit for the number of layers, it is configurable by setting `DYNAMIC_KEYMAP_LAYER_COUNT`